### DISCUSSION TITLE
:Telescope git_files doesn't show results until I arrow down

### DISCUSSION THREAD

**Discussion by lieut-data** [LunarVim/LunarVim#4587] [Q&A]
When I type to filter files using Telescope in LunarVim, I initially get unexpected results, and only after focusing the list with a down arrow down does it update:

![CleanShot 2024-07-02 at 14 59 36](https://github.com/LunarVim/LunarVim/assets/1023171/1374012c-c36f-456a-b0d9-5f926ac7f1ae)

The above is with a vanilla configuration. By contrast, if I use Telescope with nvim in the same repository, the results are consistent from the get go:

![CleanShot 2024-07-02 at 15 02 19](https://github.com/LunarVim/LunarVim/assets/1023171/06e7e517-b26f-47da-b5db-ce5d6425e424)

As I'm new to LunarVim, I'm wondering if perhaps I've just setup something wrong?

**Comment by kaykurokawa** [comment]
I have a similar problem , Telescope git_files behavior just seems quite odd to me.

First of all,  it returns non git tracked files as well as git tracked files, so that seems like a counterintuitive behavior. Maybe that's the default way it works but I would like for it to just return the files tracked by git. 

Second, for some reason if there is a large number of files in the folder, it is extra slow and behaves as @lieut-data describes. More so then just running :Telescope find_files which is extra strange. If find_files takes one second, git_files would take about 5 seconds.

**Comment by kad1kad** [comment]
I'm having the same issue. Have you found a solution @lieut-data ?

**Comment by thebrianemory** [comment]
@lieut-data @kad1kad Been dealing with the same issue for a while and finally starting looking into it. Found this helpful comment, https://github.com/LunarVim/LunarVim/issues/4642#issuecomment-2713774881, and changing the commit hash and updating plugins fixed the issue for me. Latest Telescope commit hash from https://github.com/nvim-telescope/telescope.nvim/commits/master/ is currently `"commit": "a4ed825"`.

**Reply by marija-so** [reply]
**Oh my god,** thank you for this - I spent 2 days trying to figure out why my Find File (leader+f) was **garbage** after getting a new computer. I knew it was telescope/fzf/something in that vein and I have been hunting through lua config for anything different! This was all I needed.
